drivers/storage/portworx/portworx_test.go

```diff
@@ -829,6 +829,55 @@ func TestStorageClusterDefaultsForNodeSpecs(t *testing.T) {
 	require.Equal(t, "node-kvdb", *cluster.Spec.Nodes[0].Storage.KvdbDevice)
 }
 
+func TestStorageClusterDefaultsForSecurity(t *testing.T) {
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	driver := portworx{}
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Image: "px/image:2.1.5.1",
+		},
+	}
+
+	// Security spec should be nil, as it's disabled by default
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.Nil(t, cluster.Spec.Security)
+
+	// when security.enabled is false, no security fields should be populated.
+	cluster.Spec.Security = &corev1alpha1.SecuritySpec{
+		Enabled: false,
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.Nil(t, cluster.Spec.Security.Auth)
+
+	// Check for default values when only enabled=true
+
+	cluster.Spec.Security = &corev1alpha1.SecuritySpec{
+		Enabled: true,
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.NotNil(t, cluster.Spec.Security)
+	require.Equal(t, true, cluster.Spec.Security.Enabled)
+	require.NotNil(t, true, cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer)
+	require.Equal(t, "portworx.io", *cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer)
+	require.NotNil(t, true, cluster.Spec.Security.Auth.Authenticators.SelfSigned.TokenLifetime)
+	require.Equal(t, metav1.Duration{Duration: 24 * time.Hour}, *cluster.Spec.Security.Auth.Authenticators.SelfSigned.TokenLifetime)
+
+	// issuer, when manually set, is not overwritten.
+	cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer = stringPtr("myissuer.io")
```

Comment:
Let's add the following test cases as well:
- spec.auth is not nil but empty
- spec.auth.authenticators is not nil but empty
- spec.auth.authenticators.selfSigned is not nil but empty
- issuer is not set but the issue env is present in StorageCluster.

Tip: I generally check codecov to see if there is any untested code path. It's quite handy.
https://codecov.io/gh/libopenstorage/operator/pull/168/diff?src=pr&el=tree#diff-ZHJpdmVycy9zdG9yYWdlL3BvcnR3b3J4L3BvcnR3b3J4Lmdv